File: Iteration-2/src/Advisor.java
package Projee;

import java.util.ArrayList;
import java.util.List;

public class Advisor extends Lecturer {
    private String password;
    private List<Student> students;

    public Advisor(String firstName, String lastName, String password, List<Course> assignedCourses, List<Student> students) {
        super(firstName, lastName, assignedCourses);
        this.password = password;
        this.students = new ArrayList<>();
        this.students.addAll(students);
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public List<Student> getStudents() {
        return students;
    }

    public void setStudents(List<Student> students) {
        this.students = students;
    }

    public boolean hasStudent(Student student) {
        return students.contains(student);
    }
}

File: Iteration-3/Src/Advisor.py
from Lecturer import Lecturer

class Advisor(Lecturer):
    def __init__(self, first_name, last_name, password, assigned_courses, students):
        super().__init__(first_name, last_name, assigned_courses)
        self.password = password
        self.students = students

    def get_password(self):
        return self.password

    def set_password(self, password):
        self.password = password

    def get_students(self):
        return self.students

    def set_students(self, students):
        self.students = students

    def has_student(self, student):
        return student in self.students
